tensorflow/core/kernels/gpu_utils.h

```diff
@@ -93,15 +92,26 @@ class AutotuneMap {
   bool Find(const Parameters& params, Config* config) const {
     mutex_lock lock(mu_);
     auto iter = params_config_map_.find(params);
-    if (iter == params_config_map_.end() ||
-        (iter->second.score < min_score_threshold_ &&
-         iter->second.count <= max_autotune_count_)) {
+
+    if (iter == params_config_map_.end()) {
       return false;
     }
     *config = iter->second.config;
     return true;
   }
+
   void Insert(const Parameters& params, const Config& config) {
```

Comment:
A lot of the logic in `Insert` has been moved to `InsertBasedOnScore`, but the latter function is never called. Was this logic important?

Reply:
Thanks, Reed. Can you see if the changes I made resolve the problem with your test?

It looks like the rollback is confusing github.

Reply:
@SandSnip3r was looking at the test, not me (although I think now a third person is looking at it). I happened to notice some logic in `Insert` was removed, but I don't think it was affecting the test. Anyway, the test is still being worked on, and we hope we can have an update soon.

It looks like you reverted most the PR. @SandSnip3r has been using the version of the PR when it was originally submitted to debug the test. Are any of the updates to the PR since then important?

Reply:
And by "submitted"  in the comment above, I mean "merged". @SandSnip3r is testing with https://github.com/tensorflow/tensorflow/commit/e1ba3cdfaac0010d6069d85b2e66cee659089ff7, which has since been rolled back. (@SandSnip3r, correct me if I'm wrong)

Reply:
The only update since the original merge was the removal of `InsertBasedOnScore` yesterday.

Github seems to see only the changes after the merge. It might be best to continue with a new PR.

Reply:
@philipphack, you're welcome to make a new PR if you want, but I think it is necessary unless you're introducing new non-trivial changes. Your original PR was merged, and then rolled back due to performance issues. I have a pending roll back of the rollback once we resolve the issue. There are a handful of edits that I've made in order to make everything compliant with Google style guides and build rules. If it matters to you, I think the rollback of the rollback would no longer attribute the work to your username.

As @reedwm mentioned, we have another person looking into this specific failing test which originally motivated the rollback; not everyone internally has access to this test. I'll be sure to update you on any findings.

Reply:
Re-merging based on commit e1ba3cd is probably not the best way forward, since the autotuning logic should be fixed. Although this would otherwise be resolved in the planned follow-up PR on integrating cuBLASLt into native TF, it may negatively affect performance in the meantime.

Reply:
What is the "autotuning logic" fix?

Reply:
Function `Insert` in n [gpu_utils.h](https://github.com/tensorflow/tensorflow/blob/2bbc43325f9fedd229527637ac85b1bc31c2a2aa/tensorflow/core/kernels/gpu_utils.h) was originally replaced by `InsertBasedOnScore`. However, after removing related changes due to the split of the PR into two parts, this modification effectively changes the behavior of the autotuner.

I've submitted PR #55351 which also makes it clear that cuBLASLt is being integrated into XLA only.